viewer/packages/tools/src/Measurement/MeasurementDistance.ts

```diff
@@ -0,0 +1,179 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+import { Cognite3DViewer } from '@reveal/core';
+import * as THREE from 'three';
+import { LineGeometry } from './LineGeometry';
+import { lineShaders } from '@reveal/rendering';
+import { Line } from './Line';
+import { Measurement } from './Measurement';
+import { MeasurementLineOptions } from './types';
+
+export class MeasurementDistance implements Measurement {
```

Comment:
This class has too many responsibilities:
1. Hold state of measurement
2. Create geometry
3. Determine position of endpoint while user is moving the mouse cursor
4. Keep track of "active"
5. Manage events